power_specs.py

```diff
@@ -7,941 +7,986 @@
-                line=np.array([])
-                legname=np.array([])
-                if np.size(zzz) == 0:
-                        zzz=self.Zz
-                for zz in zzz:
-                        err=self.Get_Error_z(Sim,bstft, box,derivs,params,zz)
-                        if colour != "":
-                                line=np.append(line,plt.semilogx(self.Getkbins(),err,color=colour))
-                        else:
-                                line=np.append(line,plt.semilogx(self.Getkbins(),err))
-                        legname=np.append(legname,"z="+str(zz))  
-                if title != "":
-                        plt.title(title)
-                if ylabel != "":
-                        plt.ylabel(ylabel)
-                plt.xlabel(r"$k\; (h\,\mathrm{Mpc}^{-1})$")
-                plt.semilogx(self.knotpos,np.ones(len(self.knotpos)),"ro")
-                if ymax != 0 and ymin !=0:
-                        plt.ylim(ymin,ymax)
-                plt.xlim(self.Getkbins()[0]*0.95, 4)
-                if legend:
-                        plt.legend(line, legname,bbox_to_anchor=(0., 0, 1., .25), loc=3,ncol=3, mode="expand", borderaxespad=0.)
-
-""" A class written to store the various methods related to calculating of the flux derivatives and plotting of the flux power spectra"""
+        line=np.array([])
+        legname=np.array([])
+        if np.size(zzz) == 0:
+            zzz=self.Zz
+        for zz in zzz:
+            err=self.Get_Error_z(Sim,bstft, box,derivs,params,zz)
+            if colour != "":
+                line=np.append(line,plt.semilogx(self.Getkbins(),err,color=colour))
+            else:
+                line=np.append(line,plt.semilogx(self.Getkbins(),err))
+            legname=np.append(legname,"z="+str(zz))  
+        if title != "":
+            plt.title(title)
+        if ylabel != "":
+            plt.ylabel(ylabel)
+        plt.xlabel(r"$k\; (h\,\mathrm{Mpc}^{-1})$")
+        plt.semilogx(self.knotpos,np.ones(len(self.knotpos)),"ro")
+        if ymax != 0 and ymin !=0:
+            plt.ylim(ymin,ymax)
+        plt.xlim(self.Getkbins()[0]*0.95, 4)
+        if legend:
+            plt.legend(line, legname,bbox_to_anchor=(0., 0, 1., .25), loc=3,ncol=3, mode="expand", borderaxespad=0.)
+
+    def Getkbins(self): #copy pasted from flux_pow
+        """Get the kbins to interpolate onto"""
+        flux_pow.Getkbins(self.sdsskbins)
+        #return self.sdsskbins
+    def loadpk(self, path, box): #copy pasted from flux_pow
+        """Load a Pk. Different function due to needing to be different for each class"""
+        # #Adjust Fourier convention.
+        # flux_power=np.loadtxt(self.base+path)
+        # scale=self.H0/box
+        # k=(flux_power[1:,0]-0.5)*scale*2.0*math.pi
+        # PF=flux_power[1:,1]/scale
+        # return (k, PF)
+        flux_pow.loadpk(path, box) # ASSUMING SAME FUNCTION AS "flux_pow"
```

Comment:
This should be "return flux_pow.loadpk(path, box)". You can also use "super().loadpk(path, box)" to future proof your class inheritance